Dockerfile.template

```
@@ -89,8 +93,10 @@ RUN set -eux; \
 		USE_LUA=1 LUA_INC=/usr/include/lua5.3 {{ if env.variant == "alpine" then "LUA_LIB=/usr/lib/lua5.3 " else "" end }}\
 		USE_OPENSSL=1 \
 		USE_PCRE2=1 USE_PCRE2_JIT=1 \
+{{ if ([ "1.7", "1.8", "2.0", "2.2", "2.3" ] | index(env.version)) then ( -}}
 		USE_ZLIB=1 \
-{{ if env.version | startswith("2.") and ([ "2.0", "2.2", "2.3" ] | index(env.version) | not) then ( -}}
+{{ ) else "" end -}}
+{{ if ([ "1.7", "1.8", "2.0", "2.2", "2.3" ] | index(env.version) | not) then ( -}}
 		USE_PROMEX=1 \
 {{ ) else "" end -}}
```

Since this is just the same list, we should just swap the else to use the block and then adjust the `end` block to match

```suggestion
{{ ) else ( -}}
		USE_PROMEX=1 \
{{ ) end -}}
```

---

I've intentionally split these into 2 conditions, because I didn't want to create inter-dependencies on unrelated options. It might require untangling later if for whatever reason something else changes.

Feel free to simply commit into my branch, though. Pushes are open for maintainers.

---

That's fair IMO :+1: